docs/heap/heapBasics.md

```diff
@@ -0,0 +1,190 @@
+# Heap Data Structure
+
+A **Heap** is a specialized tree-based data structure that satisfies the **Heap Property**. Heaps are commonly used to implement priority queues and ensure efficient retrieval of the minimum or maximum element.
+
+## Table of Contents
```

Comment:
deleted `Table of Contents` because already set on the right side based on `.md` files `##` or `###` tags